File: Colosus.Entity/Concretes/RequestParameter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Colosus.Entity.Concretes
{
    public class RequestParameter
    {

        public RequestParameter()
        {
            CreateDate = DateTime.Now;
            Supply = 1;
        }
        public DateTime CreateDate { get; private set; }
        public string? Token { get; set; }
        public string? RequestToken { get; set; }
        public int? Supply { get; set; }
        public string Data { get; set; }
        public string RequestParameterHash { get; set; }
        public string? Address { get; set; }
    }
}

File: Colosus.Client.Angular/src/models/RequestParameter.ts
import { sha256 } from 'js-sha256';


export class RequestParameter {
  Token?: string;
  RequestToken?: string;
  Supply?: number;
  Address?: string;

  CreateDate: Date;

  Data: string;
  RequestParameterHash: string;

  constructor(Data: string) {
    this.Data = Data;
    this.CreateDate = new Date();
    this.RequestParameterHash = sha256('test');
    this.Token = '';
    this.RequestToken = '';
  }

  toObject(): object {
    return {
      Token: this.Token,
      RequestToken: this.RequestToken,
      Supply: this.Supply,
      Data: this.Data,
      RequestParameterHash: this.RequestParameterHash,
      Address: this.Address,
    };
  }
}
